qingcloud/config.go

```diff
@@ -36,16 +42,31 @@ type QingCloudClient struct {
 	volume        *qc.VolumeService
 	loadbalancer  *qc.LoadBalancerService
 	tag           *qc.TagService
-	cache         *qc.CacheService
-	mongo         *qc.MongoService
 }
 
 func (c *Config) Client() (*QingCloudClient, error) {
 	cfg, err := config.New(c.ID, c.Secret)
+	if err != nil {
+		return nil, err
+	}
 	cfg.LogLevel = "debug"
+	qcUrl, err := url.Parse(c.EndPoint)
+	if err != nil {
+		return nil, err
+	}
+	if !strings.Contains(qcUrl.Host, ":") {
```

Comment:
I'm wondering if we could just have three separate parameters because there is a default configuration loaded when a new config object is created.